lib/sensu/server/process.rb

```diff
@@ -572,6 +575,58 @@ def determine_stale_clients
         end
       end
 
+      # Determine stale checks, those that have not sent an event
+      # in a specified amount of time (thresholds). This method
+      # iterates through the result registry. If the time since
+      # the latest event is equal to or greater than a threshold,
+      # the check `:output` is set to a descriptive message, and
+      # `:status` is set to the appropriate non-zero value, and an
+      # alert is sent. If a check has been sending events, then
+      # no alert is published.
+      def determine_stale_checks
+        @logger.info("determining stale checks")
+        @redis.smembers("clients") do |clients|
+          clients.each do |client_name|
+            @redis.keys("result:" + client_name + ":*") do |results|
+              results.each do |result_name|
+                next if result_name.split(':')[2] == 'keepalive'
```

Comment:
I think using mapped_hmset would be better than writing JSON as a string to redis. I did it this way as this is how it's done in other parts of the app.